### PR TITLE
feat: Create unit test for StartState #510

### PR DESCRIPTION
**Pull request by stefan-stefanooov**
**Description**:
Create unit test for StartState

**Related issue(s)**:
[#510](https://github.com/hashgraph/hedera-local-node/issues/510)

### CODE REVIEW

**Comment by stefan-stefanooov** [comment]
> ## [Quality Gate Passed](https://sonarcloud.io/dashboard?id=hashgraph_hedera-local-node&pullRequest=512) **Quality Gate passed**
> The SonarCloud Quality Gate passed, but some issues were introduced.
> 
> [7 New issues](https://sonarcloud.io/project/issues?id=hashgraph_hedera-local-node&pullRequest=512&resolved=false&inNewCodePeriod=true) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=hashgraph_hedera-local-node&pullRequest=512&resolved=false&inNewCodePeriod=true) No data about Coverage [0.0% Duplication on New Code](https://sonarcloud.io/component_measures?id=hashgraph_hedera-local-node&pullRequest=512&metric=new_duplicated_lines_density&view=list)
> 
> [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=hashgraph_hedera-local-node&pullRequest=512)

Handled

File: test/unit/testBed.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+function generateShellStubs(sandbox: sinon.SinonSandbox) {
+    const shellCDStub = sandbox.stub(shell, 'cd');
+    const shellExecStub = sandbox.stub(shell, 'exec');
+
+    return {
+        shellStubs :{
+            shellCDStub,
+            shellExecStub
+    }}
+}
```
Comment (georgi-l95) [inline test/unit/testBed.ts:104]: end of line and licence header

File: test/unit/states/startState.spec.ts [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+        await startState.onStart();
+
+        testSandbox.assert.calledOnce(dockerService.dockerComposeUp);
+        testSandbox.assert.match(observerSpy.callCount, 1);
+        testSandbox.assert.match(observerSpy.args[0], EventType.UnknownError);
+        testSandbox.assert.notCalled(loggerService.error);
+    })
+
+});
```
Comment (georgi-l95) [inline test/unit/states/startState.spec.ts:158]: end of line row

Comment (stefan-stefanooov) [reply test/unit/states/startState.spec.ts:158]: @georgi-l95 The next МР handles this.

File: test/unit/testBed.ts (hunk 1 above)
Comment (stefan-stefanooov) [reply test/unit/testBed.ts:104]: @georgi-l95 Will handle that in my Current MR

**Comment by stefan-stefanooov** [comment]
> Handled in a later request. Please ignore

Handled in a later request. Please ignore

**Review by georgi-l95 — DISMISSED** [review]

File: test/unit/states/StopState.spec.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
+
```
Comment (konstantinabl) [inline test/unit/states/StopState.spec.ts:1]: Probably should remove this ?

File: test/unit/states/startState.spec.ts [hunk 4]
```diff
@@ -0,0 +1,158 @@
+/*-
```
Comment (konstantinabl) [inline test/unit/states/startState.spec.ts:1]: Change the name of the class to start with capital letter

File: test/unit/states/startState.spec.ts [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+        const observer = {
+            update: observerSpy
+        }
+        startState.subscribe(observer);
+    });
+
+    afterEach(() => {
+        testSandbox.resetHistory();
+        observerSpy.resetHistory();
+        dockerService.dockerComposeUp.reset()
```
Comment (konstantinabl) [inline test/unit/states/startState.spec.ts:85]: add semicolon in the end

File: test/unit/testBed.ts [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+        ...generateShellStubs(sandbox),
+        ...generateLocalNodeStubs(sandbox, cliServiceArgs),
+    }
+
+    return testBead;
+}
+
+function resetTestBedHistory(testBead: LocalNodeTestBed) {
+    testBead.sandbox.resetHistory();
+    // testBead.loggerServiceStub.trace.resetHistory();
```
Comment (konstantinabl) [inline test/unit/testBed.ts:49]: Some leftout comments here

File: test/unit/testBed.ts [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+        processCWDStub: sinon.SinonStub;
+        processPlatformStub: sinon.SinonStub
+    },
+    shellStubs: {
+        shellCDStub: sinon.SinonStub;
+        shellExecStub: sinon.SinonStub;
+    }
+}
+
+let testBead: LocalNodeTestBed
```
Comment (konstantinabl) [inline test/unit/testBed.ts:26]: semicolon in the end, also did you mean to name it testBed?

**Review by konstantinabl — CHANGES_REQUESTED** [review]

**Review by georgi-l95 — APPROVED** [review]

**Review by konstantinabl — APPROVED** [review]

**Pull request merged**